test/integration/parachain/staging/sequential/loans.test.ts

```diff
@@ -0,0 +1,311 @@
+            addMarketExtrinsic,
+            activateMarketExtrinsic,
+        ]);
+
+        const [eventFound] = await Promise.all([
+            waitForEvent(sudoInterBtcAPI, sudoInterBtcAPI.api.events.sudo.Sudid, false, approx10Blocks),
+            api.tx.sudo.sudo(addMarketAndActivateExtrinsic).signAndSend(sudoAccount),
+        ]);
+        expect(
+            eventFound,
+            `Sudo event to create new market not found - timed out after ${approx10Blocks} ms`
+        ).to.be.true;
+    });
+
+    after(async () => {
+        api.disconnect();
+    });
+
+    afterEach(() => {
+        // discard any stubbed methods after each test
+        sinon.restore();
+    });
+
+    describe("getLendTokens", () => {
+        it("should get lend token for each existing market", async () => {
+            const [markets, lendTokens] = await Promise.all([
+                api.query.loans.markets.entries(),
+                userInterBtcAPI.loans.getLendTokens(),
+            ]);
+
+            const marketsUnderlyingCurrencyId = markets[0][0].args[0];
+
+            expect(markets.length).to.be.equal(lendTokens.length);
+
+            expect(marketsUnderlyingCurrencyId.eq(underlyingCurrencyId)).to.be.true;
+        });
+
+        it("should return LendToken in correct format - 'q' prefix, correct id", async () => {
+            // Requires first market to be initialized for governance currency.
+            const lendTokens = await userInterBtcAPI.loans.getLendTokens();
+            const lendToken = lendTokens[0];
+
+            // Should have same amount of decimals as underlying currency.
+            expect(lendToken.decimals).to.be.eq(underlyingCurrency.decimals);
+
+            // Should add 'q' prefix.
+            expect(lendToken.name).to.be.eq(`q${underlyingCurrency.name}`);
+            expect(lendToken.ticker).to.be.eq(`q${underlyingCurrency.ticker}`);
+
+            expect(lendToken.lendToken.id).to.be.eq(lendTokenId.asLendToken.toNumber());
+        });
+
+        it("should return empty array if no market exists", async () => {
+            // Mock empty list returned from chain.
+            sinon.stub(LoansAPI, "getLoansMarketsEntries").returns(Promise.resolve([]));
+
+            const lendTokens = await LoansAPI.getLendTokens();
+            expect(lendTokens).to.be.empty;
+
+            sinon.restore();
+            sinon.reset();
+        });
+    });
+
+    describe("getLendPositionsOfAccount", () => {
+        let lendAmount: MonetaryAmount<CurrencyExt>;
+        before(async function () {
+            this.timeout(approx10Blocks);
+
+            lendAmount = newMonetaryAmount(1, underlyingCurrency, true);
+            const lendExtrinsic = api.tx.loans.mint(underlyingCurrencyId, lendAmount.toString(true));
+
+            const [eventFound] = await Promise.all([
+                waitForEvent(userInterBtcAPI, api.events.loans.Deposited, false, approx10Blocks),
+                lendExtrinsic.signAndSend(userAccount),
```

Comment:
any reason you're not using the loans API and instead making the polkadot-js call directly in here?

Reply:
I think I added this test before adding the actual implementation for lending.